client/src/components/EventDetail.test.tsx

```diff
@@ -1,23 +1,24 @@
 import { render, screen } from "@testing-library/react";
-import { MemoryRouter } from "react-router-dom";
+import { MemoryRouter, Route, Routes } from "react-router-dom";
 import EventDetail from "./EventDetail";
-import { vi, beforeEach, afterEach, describe, it, expect } from "vitest";
+import type { components } from "../types";
 
-// Přetížení useParams tak, aby vracel id = "1"
-vi.mock("react-router-dom", async (orig) => {
-    const actual = await vi.importActual<typeof import("react-router-dom")>(
-        "react-router-dom"
-    );
-    return {
-        ...actual,
-        useParams: () => ({ id: "1" }),
-    };
-});
+// Mock vygenerovaného API klienta
+vi.mock("../api", () => ({
+    EventsService: {
+        listEvents: vi.fn(),
+        getEventById: vi.fn(),
+        createEvent: vi.fn(),
+    },
+}));
+import { EventsService } from "../api";
```

Comment:
co je EventsService? Nemuzu to nikde najit :) Stejne tak mi to nenajde, co je
```
import type { components } from "../types";
```

<img width="1504" height="272" alt="image" src="https://github.com/user-attachments/assets/b8838d40-5f2b-40e7-ae65-d608b5606e82" />

Reply:
tohle souvisi s tim konfliktem v nazvu souborech :) Kdyz svuj prejmenujes, bude to opravene